joomla/joomla-cms
# Improve Datetime validation

## Problem

Currently, when you enter an invalid datetime value for a calendar form field, Joomla will throw a system error like this

> DateTime::__construct() [datetime.--construct]: Failed to parse time string (jerror) ......

There were atleast two issues reported #8646 and #8093 which I can see
## Solution

I attempt to fix this issue by introducing a new form rule JFormRuleDatetime which uses strtotime($value) === false command to validate the value of the field. If it return false, the date is invalid and an exception will be returned
## Testing instructions

With this PR, I just modify the add/edit article form in the backend of Joomla to show how it works.

Please try to edit an article, before this patch, if you enter a wrong value for any Calendar field (for example Start Publishing, Finish Publishing.., you will see a system error

After this patch, there will be an error message like this being showed and it will prevent you from saving the article

> Warning
> Value 2015-13-30 19:07:12 you entered for Start Publishing is invalid datetime format
